src/evernodetwo.c

```diff
@@ -668,6 +672,64 @@ int64_t hook(uint32_t reserved)
 
             accept(SBUF("Evernode: Dead Host Pruning successful."), 0);
         }
+        else if (is_host_transfer)
+        {
+            // Check for registration entry, if transferee is different from transfer (transferring to a new account).
+            int is_host_as_transferee = 0;
+            BUFFER_EQUAL(is_host_as_transferee, data_ptr, account_field, 20);
+            if (is_host_as_transferee == 0)
+            {
+                HOST_ADDR_KEY(data_ptr); // Generate account key for transferee.
+                uint8_t reg_entry_buf[HOST_ADDR_VAL_SIZE];
+                if (state(SBUF(reg_entry_buf), SBUF(STP_HOST_ADDR)) != DOESNT_EXIST)
+                    rollback(SBUF("Evernode: New transferee also a registered host."), 1);
+            }
+
+            // Check whether this host has an initiated transfer.
+            uint8_t host_transfer_flag = host_addr[HOST_TRANSFER_FLAG_OFFSET];
+            if (host_transfer_flag == PENDING_TRANSFER)
+                rollback(SBUF("Evernode: Host has a pending transfer."), 1);
+
+            // Check whether there is an already initiated transfer for the transferee
+            TRANSFEREE_ADDR_KEY(data_ptr);
+            // <transferring_host_address(20)><registration_ledger(8)><token_id(20)>
+            uint8_t transferee_addr[TRANSFEREE_ADDR_VAL_SIZE];
+            CLEAR_BUF(transferee_addr, 0, TRANSFEREE_ADDR_VAL_SIZE); // Initialize buffer wih 0s
+
+            if (state(SBUF(transferee_addr), SBUF(STP_TRANSFEREE_ADDR)) != DOESNT_EXIST)
+                rollback(SBUF("Evernode: There is a previously initiated transfer for this transferee."), 1);
+
+            // Saving the Pending transfer in Hook States.
+            COPY_BUF(transferee_addr, TRANSFER_HOST_ADDRESS_OFFSET, account_field, 0, ACCOUNT_ID_SIZE);
+            INT64_TO_BUF(&transferee_addr[TRANSFER_HOST_LEDGER_OFFSET], cur_ledger_seq);
+
+            uint8_t *token_id_ptr = &host_addr[HOST_TOKEN_ID_OFFSET];
+            COPY_BUF(transferee_addr, TRANSFER_HOST_TOKEN_ID_OFFSET, token_id_ptr, 0, NFT_TOKEN_ID_SIZE);
```

Comment:
We can give like `COPY_BUF(transferee_addr, TRANSFER_HOST_TOKEN_ID_OFFSET, host_addr, HOST_TOKEN_ID_OFFSET, NFT_TOKEN_ID_SIZE);` too